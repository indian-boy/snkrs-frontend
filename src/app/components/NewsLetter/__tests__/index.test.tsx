import { render } from '@testing-library/react';
import React from 'react';
import { NewsLetter } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<NewsLetter  />', () => {
  it('should match snapshot', () => {
    const newsLetter = render(<NewsLetter />);
    expect(newsLetter.container.firstChild).toMatchSnapshot();
  });
});
