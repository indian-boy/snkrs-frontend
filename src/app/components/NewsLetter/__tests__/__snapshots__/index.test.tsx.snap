// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsLetter  /> should match snapshot 1`] = `
.c3 {
  padding: 0.9rem 1.5rem;
  border-radius: 3.125rem;
  font-style: normal;
  font-weight: bold;
  font-size: 1rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  outline: none;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background-color: #000000;
  color: #FFFFFF;
  box-shadow: 0 0.2rem 0.5rem rgba(0,0,0,0.1);
}

.c3:focus,
.c3:active {
  outline: none;
}

.c3:hover,
.c3:focus,
.c3:active {
  background-color: #FFFFFF;
  color: #000000;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #F7F7F7;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4rem 1rem;
}

.c1 {
  font-style: normal;
  font-weight: bold;
  font-size: 1.125rem;
  line-height: 1.2rem;
  color: #1A1A1A;
  margin-bottom: 0.5rem;
  text-transform: uppercase;
  text-align: center;
}

.c2 {
  font-style: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #1A1A1A;
  margin-bottom: 1rem;
  text-align: center;
}

@media (min-width:1024px) {
  .c0 {
    padding: 4rem 2.5rem;
  }
}

<div
  class="c0"
>
  <svg
    style="margin-bottom: 1.5rem;"
  >
    nike.svg
  </svg>
  <span
    class="c1"
  >
    components.newsLetter.title
    RECEBA AS NOVIDADES DA NIKE
  </span>
  <span
    class="c2"
  >
    components.newsLetter.message
    Cadastre-se e seja um dos primeiros a saber de todas as novidades e ofertas.
  </span>
  <button
    class="c3"
    type="button"
  >
    components.newsLetter.registerButtonLabel
    Cadastre-se
  </button>
</div>
`;
